src/components/AttachmentCarousel/AttachmentCarouselView/index.js

```diff
@@ -0,0 +1,152 @@
+            // so we only enable it for mobile
+            scrollRef.current.scrollToIndex({index: nextIndex, animated: false});
+        },
+        [props.carouselState.attachments, props.carouselState.page],
+    );
+
+    /**
+     * Defines how a container for a single attachment should be rendered
+     * @param {Object} props
+     * @returns {JSX.Element}
+     */
+    const renderCell = useCallback((cellProps) => {
+        // Use window width instead of layout width to address the issue in https://github.com/Expensify/App/issues/17760
+        // considering horizontal margin and border width in centered modal
+        const modalStyles = styles.centeredModalStyles(cellProps.isSmallScreenWidth, true);
+        const style = [cellProps.style, styles.h100, {width: PixelRatio.roundToNearestPixel(cellProps.windowWidth - (modalStyles.marginHorizontal + modalStyles.borderWidth) * 2)}];
+
+        return (
+            <View
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...cellProps}
+                style={style}
+            />
+        );
+    }, []);
+
+    /**
+     * Defines how a single attachment should be rendered
+     * @param {{ isAuthTokenRequired: Boolean, source: String, file: { name: String } }} item
+     * @returns {JSX.Element}
+     */
+    const renderItem = useCallback(
+        ({item}) => (
+            <AttachmentView
+                source={item.source}
+                file={item.file}
+                isAuthTokenRequired={item.isAuthTokenRequired}
+            />
+        ),
+        [],
+    );
+
+    return (
+        <>
+            <CarouselButtons
+                carouselState={props.carouselState}
+                onBack={() => {
+                    cycleThroughAttachments(-1);
+                    props.autoHideArrow();
+                }}
+                onForward={() => {
+                    cycleThroughAttachments(1);
+                    props.autoHideArrow();
+                }}
+                autoHideArrow={props.autoHideArrow}
+                cancelAutoHideArrow={props.cancelAutoHideArrow}
+            />
+
+            {props.carouselState.containerWidth > 0 && (
+                <FlatList
+                    listKey="AttachmentCarousel"
+                    horizontal
+                    decelerationRate="fast"
+                    showsHorizontalScrollIndicator={false}
+                    bounces={false}
+                    // Scroll only one image at a time no matter how fast the user swipes
+                    disableIntervalMomentum
+                    pagingEnabled
+                    snapToAlignment="start"
+                    snapToInterval={props.carouselState.containerWidth}
+                    scrollEnabled={false}
+                    ref={scrollRef}
+                    initialScrollIndex={props.carouselState.page}
+                    initialNumToRender={3}
+                    windowSize={5}
```

Comment:
Where are these hardcoded numbers coming from? Or are they just WIP?

Reply:
Oh I see, they already existed. Feel free to resolve

Reply:
That's all from `main`, we didn't change that